modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -2975,6 +2976,85 @@ void ONNXImporter::parseCumSum(LayerParams& layerParams, const opencv_onnx::Node
     addLayer(layerParams, node_proto);
 }
 
+void ONNXImporter::parseDepthToSpace(LayerParams& layerParams, const opencv_onnx::NodeProto& node_proto_)
+{
+    // We parse "DepthToSpace" and "SpaceToDepth" in this function.
+    opencv_onnx::NodeProto node_proto = node_proto_;
+    const std::string& layer_type = node_proto.op_type();
+    CV_Assert(layer_type == "DepthToSpace" || layer_type == "SpaceToDepth");
+    layerParams.type = layer_type;
+
+    // Get blocksize
+    int blocksize = 0;
+    if(layerParams.has("blocksize"))
+        blocksize = layerParams.get<int>("blocksize");
+    CV_CheckGT(blocksize, 0, "");
+
+    // Get mode, only for "DepthToSpace"
+    std::string modeType = "DCR";
+    if(layerParams.has("mode"))
+        modeType = layerParams.get<std::string>("mode");
+
+    MatShape inpShape = outShapes[node_proto.input(0)];
+    CV_Assert(inpShape.size() == 4);
+
+    // Implement DepthToSpace and SpaceToDepth by the Reshape and Permute layer.
+    std::vector<int> shape0, shape1, perm;
+    if(layer_type == "DepthToSpace")
+    {
+        if(modeType == "DCR")
+        {
+            shape0 = {inpShape[0], blocksize, blocksize, inpShape[1]/(blocksize * blocksize), inpShape[2], inpShape[3]};
```

Comment:
I'm not a fan of inpShape[i], you could name each dimension(N, C, H, W), if you want. But it's fine as is.